packages/flutter/lib/src/material/slider_theme.dart

```diff
@@ -1683,18 +1683,46 @@ class RoundedRectSliderTrackShape extends SliderTrackShape with BaseSliderTrackS
     );
 
     // The arc rects create a semi-circle with radius equal to track height.
-    final Rect leftTrackArcRect = Rect.fromLTWH(trackRect.left, trackRect.top, trackRect.height, trackRect.height);
+    final Rect leftTrackArcRect = Rect.fromLTWH(
+      trackRect.left - trackRect.height / 2,
+      trackRect.top,
+      trackRect.height,
+      trackRect.height,
+    );
+
+    final Rect rightTrackArcRect = Rect.fromLTWH(
+      trackRect.right - trackRect.height / 2,
+      trackRect.top,
+      trackRect.height,
+      trackRect.height,
+    );
+
+    final Rect leftTrackSegment = Rect.fromLTRB(
+      trackRect.left,
+      trackRect.top,
+      thumbCenter.dx,
+      trackRect.bottom,
+    );
+
+    final Rect rightTrackSegment = Rect.fromLTRB(
+      thumbCenter.dx,
+      trackRect.top,
+      trackRect.right,
+      trackRect.bottom,
+    );
+
+    // all drawing
     if (!leftTrackArcRect.isEmpty)
-      context.canvas.drawArc(leftTrackArcRect, math.pi / 2, math.pi, false, leftTrackPaint);
-    final Rect rightTrackArcRect = Rect.fromLTWH(trackRect.right - trackRect.height / 2, trackRect.top, trackRect.height, trackRect.height);
+      context.canvas.drawArc(leftTrackArcRect, math.pi / 2, math.pi, false,
+          leftTrackSegment.isEmpty ? rightTrackPaint : leftTrackPaint);
+
     if (!rightTrackArcRect.isEmpty)
-      context.canvas.drawArc(rightTrackArcRect, -math.pi / 2, math.pi, false, rightTrackPaint);
+      context.canvas.drawArc(rightTrackArcRect, -math.pi / 2, math.pi, false,
+          rightTrackSegment.isEmpty ? leftTrackPaint : rightTrackPaint);
 
```

Comment:
filling the right arc if the shape is full, and emptying the left if it is empty.
As it was in before, left is always filled and right is always emptyed.